modules/kit-elements-defaults/controller.php

```diff
@@ -0,0 +1,242 @@
+							return is_array( $param );
+						},
+						'sanitize_callback' => function( $param, \WP_REST_Request $request ) {
+							return $this->sanitize_settings( $param, $request );
+						},
+					],
+				],
+				'callback' => $this->wrap_callback( function ( \WP_REST_Request $request ) {
+					return $this->update( $request );
+				} ),
+				'permission_callback' => function () {
+					return current_user_can( 'manage_options' );
+				},
+			],
+		] );
+	}
+
+	private function update( \WP_REST_Request $request ) {
+		$kit = Plugin::$instance->kits_manager->get_active_kit();
+
+		$data = $kit->get_json_meta( static::META_KEY );
+
+		$data[ $request->get_param( 'type' ) ] = $request->get_param( 'settings' );
+
+		$kit->update_meta(
+			static::META_KEY,
+			wp_json_encode( $data )
+		);
+
+		return new \WP_REST_Response( [], 201 );
+	}
+
+
+	/**
+	 * DELETE '/kit-elements-defaults/{type}'
+	 */
+	private function register_destroy() {
+		register_rest_route( static::NAMESPACE, static::BASE_URL . '/(?P<type>[\w\-_]+)', [
+			[
+				'methods' => 'DELETE',
+				'args' => [
+					'type' => [
+						'type' => 'string',
+						'description' => 'The type of the element.',
+						'required' => true,
+						'validate_callback' => function( $param ) {
+							return $this->validate_type( $param );
+						},
+					],
+				],
+				'callback' => $this->wrap_callback( function ( \WP_REST_Request $request ) {
+					return $this->destroy( $request );
+				} ),
+				'permission_callback' => function () {
+					return current_user_can( 'manage_options' );
+				},
+			],
+		] );
+	}
+
+	private function destroy( \WP_REST_Request $request ) {
+		$kit = Plugin::$instance->kits_manager->get_active_kit();
+
+		$data = $kit->get_json_meta( static::META_KEY );
+
+		unset( $data[ $request->get_param( 'type' ) ] );
+
+		$kit->update_meta(
+			static::META_KEY,
+			wp_json_encode( $data )
+		);
+
+		return new \WP_REST_Response( [], 204 );
+	}
+
+	/**
+	 * Wrapper to validate that the kit is valid.
+	 * Essentially behaves as a middleware.
+	 *
+	 * @param $callback
+	 *
+	 * @return \Closure
+	 */
+	private function wrap_callback( $callback ) {
+		return function ( \WP_REST_Request $request ) use ( $callback ) {
+			$kit = Plugin::$instance->kits_manager->get_active_kit();
+
+			// TODO: Is this right?
+			$is_valid_kit = $kit && $kit->get_main_id();
+
+			if ( ! $is_valid_kit ) {
+				return new \WP_Error( 'invalid_kit', 'Kit doesn\'t exist.', [
+					'status' => 404,
+				] );
+			}
+
+			return $callback( $request );
+		};
+	}
+
+	private function validate_type( $param ) {
+		$element_types = array_keys( Plugin::$instance->elements_manager->get_element_types() );
+		$widget_types = array_keys( Plugin::$instance->widgets_manager->get_widget_types() );
+
+		return in_array(
+			$param,
+			array_merge( $element_types, $widget_types ),
+			true
+		);
+	}
+
+	private function sanitize_settings( $param, \WP_REST_Request $request ) {
```

Comment:
Maybe extract into diffrent class? + all the related methods

Reply:
Good idea. Done